pandas/core/indexes/base.py

```diff
@@ -4020,6 +4020,10 @@ def _add_numeric_methods_binary(cls):
 
         def _make_evaluate_binop(op, opstr, reversed=False, constructor=Index):
             def _evaluate_numeric_binop(self, other):
+                if (isinstance(other, ABCSeries) and
+                        self.dtype == "timedelta64[ns]"):
+                    # GH#19042
+                    return NotImplemented
```

Comment:
Explain why you're doing this.

Reply:
I'll expand the comment.  This actually needs to be done for _all_ Index classes because the problem is wide-spread, but I really want to avoid scope creep.

Reply:
Fair enough.  I agree that we should try to keep the scope well-defined, though I might consider adding tests that you can `xfail` for now and patch in a follow-up (if you can add / make tests generic enough to apply to *all* Index classes).